app/src/main/scala/org/alephium/explorer/Application.scala

```diff
@@ -103,14 +106,34 @@ class Application(
       _ <- mempoolSyncService.start(peers)
       _ <- tokenSupplyService.start()
       _ <- hashrateService.start()
+      _ <- finalizerService.start()
     } yield ()
   }
 
+  private def startReadWrite(): Future[Unit] = {
+    if (!readOnly) {
+      for {
+        _ <- dbInitializer.initialize()
+        _ <- if (readOnly) Future.successful(()) else startSyncService()
+      } yield ()
+
+    } else {
+      Future.successful(())
+    }
+  }
+
+  private def stopReadWrite(): Future[Unit] = {
+    if (!readOnly) {
+      blockFlowSyncService.stop()
+    } else {
+      Future.successful(())
+    }
+  }
+
   def start: Future[Unit] = {
```

Comment:
Future Improvements: We should either use or build a simple framework for services so they can be started and stopped properly without having to deal with dependencies manually. Full node ref: https://github.com/alephium/alephium/blob/master/util/src/main/scala/org/alephium/util/Service.scala

Reply:
https://github.com/alephium/explorer-backend/issues/201